sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java

```diff
@@ -236,4 +246,71 @@ public void testSplitForTopic() throws Exception {
     assertEquals(1, splits.size());
   }
 
+  @Test
+  public void testCheckpointMark() throws Exception {
+    Connection connection = connectionFactoryWithoutPrefetch.createConnection(USERNAME, PASSWORD);
+    connection.start();
+    Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+    MessageProducer producer = session.createProducer(session.createQueue(QUEUE));
+    for (int i = 0; i < 1000; i++) {
```

Comment:
Suppose this will still work with 10 messages rather than 1000, but faster, right? :)